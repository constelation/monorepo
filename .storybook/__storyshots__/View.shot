exports[`alignVertical center alignHorizontal right`] = `
<div
  data-css-16fd8ps="*"
  style={undefined}>
  <div
    data-css-1on9f6q="*"
    style={undefined} />
  <div
    data-css-1s71gus="*"
    style={undefined} />
  <div
    data-css-3yrej3="*"
    style={undefined} />
</div>
`;

exports[`console.logs refNode`] = `
<div
  data-css-33m34m="*"
  style={undefined} />
`;

exports[`only using inlineStyle`] = `
<div
  data-css-33m34m="*"
  style={
    Object {
      "backgroundColor": "lightgrey",
      "border": "1px solid black",
      "height": 500,
    }
  }>
  <div
    data-css-33m34m="*"
    style={
      Object {
        "backgroundColor": "red",
        "flex": 1,
      }
    } />
  <div
    data-css-33m34m="*"
    style={
      Object {
        "backgroundColor": "green",
        "flex": 1,
      }
    } />
  <div
    data-css-33m34m="*"
    style={
      Object {
        "backgroundColor": "blue",
        "flex": 1,
      }
    } />
</div>
`;

exports[`with children Views and style prop`] = `
<div
  data-css-ab0pxa="*"
  style={undefined}>
  <div
    data-css-1wwt2nx="*"
    style={undefined} />
  <div
    data-css-14nlop9="*"
    style={undefined} />
  <div
    data-css-1bbyl59="*"
    style={undefined} />
</div>
`;

exports[`with children divs and inline style`] = `
<div
  data-css-ab0pxa="*"
  style={undefined}>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "height": 20,
      }
    } />
  <div
    style={
      Object {
        "backgroundColor": "green",
        "height": 20,
      }
    } />
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "height": 20,
      }
    } />
</div>
`;

exports[`with children flex=1 Views`] = `
<div
  data-css-ab0pxa="*"
  style={undefined}>
  <div
    data-css-1wwt2nx="*"
    style={undefined} />
  <div
    data-css-14nlop9="*"
    style={undefined} />
  <div
    data-css-1bbyl59="*"
    style={undefined} />
</div>
`;
