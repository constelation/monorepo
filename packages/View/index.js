"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorReact = require("glamor-react");
var _omit = require("lodash/omit");
var alignHorizontalAlias = {
    left: 'flex-start',
    center: 'center',
    right: 'flex-end',
};
var alignVerticalAlias = {
    top: 'flex-start',
    center: 'center',
    bottom: 'flex-end',
};
var propsToOmit = [
    'align',
    'alignSelf',
    'alignContent',
    'alignHorizontal',
    'alignVertical',
    'justify',
    'bottom',
    'flex',
    'wrap',
    'grow',
    'shrink',
    'basis',
    'horizontal',
    'height',
    'left',
    'margin',
    'marginBottom',
    'marginLeft',
    'marginRight',
    'marginTop',
    'maxHeight',
    'maxWidth',
    'minHeight',
    'minWidth',
    'order',
    'overflow',
    'overflowX',
    'overflowY',
    'padding',
    'paddingBottom',
    'paddingLeft',
    'paddingRight',
    'paddingTop',
    'pointerEvents',
    'position',
    'right',
    'top',
    'width',
    'zIndex',
    'marginHorizontal',
    'marginVertical',
    'paddingHorizontal',
    'paddingVertical',
    'style',
    'inlineStyle',
    'tag',
    'overflowScrolling',
    'hidden',
    'inline',
    'fit',
    'center',
    'hitSlop',
    'hitSlopVertical',
    'hitSlopHorizontal',
    'hitSlopTop',
    'hitSlopRight',
    'hitSlopBottom',
    'hitSlopLeft',
    'refNode',
];
function getAlignItems(props) {
    if (props.align) {
        return props.align;
    }
    else if (props.horizontal === true && props.alignVertical) {
        return alignVerticalAlias[props.alignVertical];
    }
    else if (props.alignHorizontal) {
        return alignHorizontalAlias[props.alignHorizontal];
    }
    else if (props.center === true) {
        return 'center';
    }
    else if (props.horizontal === true) {
        return 'flex-start';
    }
    else {
        return 'stretch';
    }
}
function getJustifyContent(props) {
    if (props.justify) {
        return props.justify;
    }
    else if (props.horizontal === true && props.alignHorizontal) {
        return alignHorizontalAlias[props.alignHorizontal];
    }
    else if (props.alignVertical) {
        return alignVerticalAlias[props.alignVertical];
    }
    else if (props.center === true) {
        return 'center';
    }
}
function getStyleFromProps(props) {
    return {
        alignSelf: props.alignSelf,
        alignItems: getAlignItems(props),
        alignContent: props.alignContent,
        bottom: props.bottom,
        display: props.hidden ? 'none' : (props.inline ? 'inline-flex' : 'flex'),
        flex: props.flex,
        flexDirection: props.horizontal === true ? 'row' : 'column',
        flexWrap: props.wrap,
        flexGrow: props.grow === true ? 1 : props.grow,
        flexShrink: props.shrink,
        flexBasis: props.basis,
        height: props.fit ? '100%' : props.height,
        justifyContent: getJustifyContent(props),
        left: props.left,
        margin: props.margin,
        marginBottom: props.marginBottom || props.marginVertical,
        marginLeft: props.marginLeft || props.marginHorizontal,
        marginRight: props.marginRight || props.marginHorizontal,
        marginTop: props.marginTop || props.marginVertical,
        maxHeight: props.maxHeight,
        maxWidth: props.maxWidth,
        minHeight: props.minHeight,
        minWidth: props.minWidth,
        order: props.order,
        overflow: props.overflow,
        overflowX: props.overflowX,
        overflowY: props.overflowY,
        padding: props.padding,
        paddingBottom: props.paddingBottom || props.paddingVertical,
        paddingLeft: props.paddingLeft || props.paddingHorizontal,
        paddingRight: props.paddingRight || props.paddingHorizontal,
        paddingTop: props.paddingTop || props.paddingVertical,
        pointerEvents: props.pointerEvents,
        position: props.position,
        right: props.right,
        top: props.top,
        width: props.fit ? '100%' : props.width,
        WebkitOverflowScrolling: props.overflowScrolling,
        zIndex: props.zIndex,
    };
}
function hasHitSlopProp(props) {
    return props.hitSlop
        || props.hitSlopVertical
        || props.hitSlopHorizontal
        || props.hitSlopTop
        || props.hitSlopRight
        || props.hitSlopBottom
        || props.hitSlopLeft;
}
var Slop = function (props) { return (React.createElement("span", { style: {
        position: 'absolute',
        top: -(props.hitSlopTop || props.hitSlopVertical || props.hitSlop || 0),
        right: -(props.hitSlopRight || props.hitSlopHorizontal || props.hitSlop || 0),
        bottom: -(props.hitSlopBottom || props.hitSlopVertical || props.hitSlop || 0),
        left: -(props.hitSlopLeft || props.hitSlopHorizontal || props.hitSlop || 0),
    } })); };
var View = (function (_super) {
    __extends(View, _super);
    function View() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    View.prototype.render = function () {
        var styleFromProps = getStyleFromProps(this.props);
        var propsToPass = _omit(this.props, propsToOmit);
        var css = __assign({}, styleFromProps, this.props.style);
        propsToPass.css = css;
        propsToPass.style = this.props.inlineStyle;
        if (this.props.refNode) {
            propsToPass.ref = this.props.refNode;
        }
        if (hasHitSlopProp(this.props)) {
            var HitSlop = (React.createElement(Slop, { hitSlop: this.props.hitSlop, hitSlopTop: this.props.hitSlopTop, hitSlopRight: this.props.hitSlopRight, hitSlopBottom: this.props.hitSlopBottom, hitSlopLeft: this.props.hitSlopLeft, hitSlopVertical: this.props.hitSlopVertical, hitSlopHorizontal: this.props.hitSlopHorizontal }));
            if (propsToPass.children == null) {
                propsToPass.children = HitSlop;
            }
            else {
                propsToPass.children = React.Children.toArray(propsToPass.children);
                propsToPass.children.unshift(HitSlop);
            }
        }
        return glamorReact.createElement(this.props.tag, propsToPass);
    };
    return View;
}(React.PureComponent));
View.defaultProps = {
    tag: 'div',
    shrink: 0,
    position: 'relative',
    alignContent: 'flex-start',
};
exports.default = View;
