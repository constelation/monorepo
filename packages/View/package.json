{
  "name": "constelation-view",
  "version": "12.2.0",
  "description": "",
  "scripts": {
    "copy": "find ./src -name \"*.js*\" -exec cp {} . \\;",
    "build:web": "../../node_modules/.bin/tsc -p tsconfig.web.json",
    "build:native": "../../node_modules/.bin/tsc -p tsconfig.native.json",
    "build": "npm run build:web; npm run build:native; npm run copy",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [
    "constelation",
    "view"
  ],
  "author": "Kyle Poole <2kylepoole@gmail.com> (http://kylpo.com/)",
  "license": "ISC",
  "peerDependencies": {
    "lodash": "^4.13.1",
    "react": "^15.2.1"
  },
  "optionalDependencies": {
    "glamor": "^2.20.24",
    "glamor-react": "^3.0.0-1"
  }
}
