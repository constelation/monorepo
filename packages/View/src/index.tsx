import * as React from 'react'
import * as glamorReact from 'glamor-react'
import * as _omit from 'lodash/omit'

export interface IProps {
  alignContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'stretch',
  alignVertical?: 'top' | 'center' | 'bottom',
  alignHorizontal?: 'left' | 'center' | 'right',
  alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch',
  align?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch',
  justify?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around',
  bottom?: number | string,
  center?: boolean,
  fit?: boolean,

  flex?: number | string,
  wrap?: string,
  grow?: number | boolean,
  shrink?: number,
  basis?: string | number,

  height?: number | string,
  hidden?: boolean,
  hitSlop?: number | string,
  hitSlopHorizontal?: number | string,
  hitSlopVertical?: number | string,
  hitSlopTop?: number | string,
  hitSlopRight?: number | string,
  hitSlopBottom?: number | string,
  hitSlopLeft?: number | string,

  horizontal?: boolean,
  inline?: boolean,
  left?: number | string,
  margin?: number | string,
  marginBottom?: number | string,
  marginLeft?: number | string,
  marginRight?: number | string,
  marginTop?: number | string,
  marginVertical?: number | string,
  marginHorizontal?: number | string,
  maxHeight?: number | string,
  maxWidth?: number | string,
  minHeight?: number | string,
  minWidth?: number | string,
  order?: number,
  overflow?: string,
  overflowX?: string,
  overflowY?: string,
  overflowScrolling?: string,
  padding?: number | string,
  paddingBottom?: number | string,
  paddingLeft?: number | string,
  paddingRight?: number | string,
  paddingTop?: number | string,
  paddingVertical?: number | string,
  paddingHorizontal?: number | string,
  pointerEvents?: string,
  position?: string,
  refNode?: () => {},
  right?: number | string,
  inlineStyle?: Object,
  style?: Object,
  top?: number | string,
  tag?: string,
  width?: number | string,
  zIndex?: number,
}

const alignHorizontalAlias = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
}
const alignVerticalAlias = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end',
}

const propsToOmit = [
  'align',
  'alignSelf',
  'alignContent',
  'alignHorizontal',
  'alignVertical',
  'justify',
  'bottom',
  'flex',

  'wrap',
  'grow',
  'shrink',
  'basis',

  'horizontal',
  'height',
  'left',
  'margin',
  'marginBottom',
  'marginLeft',
  'marginRight',
  'marginTop',
  'maxHeight',
  'maxWidth',
  'minHeight',
  'minWidth',
  'order',
  'overflow',
  'overflowX',
  'overflowY',
  'padding',
  'paddingBottom',
  'paddingLeft',
  'paddingRight',
  'paddingTop',
  'pointerEvents',
  'position',
  'right',
  'top',
  'width',
  'zIndex',
  'marginHorizontal',
  'marginVertical',
  'paddingHorizontal',
  'paddingVertical',

  // don't want to pass style or tag down
  'style',
  'inlineStyle',
  'tag',

  // soon to be moved to aliases
  'overflowScrolling',
  'hidden',
  'inline',
  'fit',
  'center',

  'hitSlop',
  'hitSlopVertical',
  'hitSlopHorizontal',
  'hitSlopTop',
  'hitSlopRight',
  'hitSlopBottom',
  'hitSlopLeft',

  'refNode',
]

function getAlignItems(props: IProps) {
  if (props.align) {
    return props.align
  }
  else if (props.horizontal === true && props.alignVertical) {
    return alignVerticalAlias[props.alignVertical]
  }
  else if (props.alignHorizontal) {
    return alignHorizontalAlias[props.alignHorizontal]
  }
  else if (props.center === true) {
    return 'center'
  }
  else if (props.horizontal === true) {
    return 'flex-start'
  }
  else {
    return 'stretch'
  }
}

function getJustifyContent(props: IProps) {
  if (props.justify) {
    return props.justify
  }
  else if (props.horizontal === true && props.alignHorizontal) {
    return alignHorizontalAlias[props.alignHorizontal]
  }
  else if (props.alignVertical) {
    return alignVerticalAlias[props.alignVertical]
  }
  else if (props.center === true) {
    return 'center'
  }
}

function getStyleFromProps(props: IProps) {
  // some defaults from https://github.com/facebook/css-layout#default-values
  return {
    alignSelf: props.alignSelf,
    alignItems: getAlignItems(props),
    alignContent: props.alignContent,
    bottom: props.bottom,
    display: props.hidden ? 'none' : (props.inline ? 'inline-flex' : 'flex'),
    flex: props.flex,
    flexDirection: props.horizontal === true ? 'row' : 'column',
    flexWrap: props.wrap,
    flexGrow: props.grow === true ? 1 : props.grow,
    flexShrink: props.shrink,
    flexBasis: props.basis,
    height: props.fit ? '100%' : props.height,
    justifyContent: getJustifyContent(props),
    left: props.left,
    margin: props.margin,
    marginBottom: props.marginBottom || props.marginVertical,
    marginLeft: props.marginLeft || props.marginHorizontal,
    marginRight: props.marginRight || props.marginHorizontal,
    marginTop: props.marginTop || props.marginVertical,
    maxHeight: props.maxHeight,
    maxWidth: props.maxWidth,
    minHeight: props.minHeight,
    minWidth: props.minWidth,
    order: props.order,
    overflow: props.overflow,
    overflowX: props.overflowX,
    overflowY: props.overflowY,
    padding: props.padding,
    paddingBottom: props.paddingBottom || props.paddingVertical,
    paddingLeft: props.paddingLeft || props.paddingHorizontal,
    paddingRight: props.paddingRight || props.paddingHorizontal,
    paddingTop: props.paddingTop || props.paddingVertical,
    pointerEvents: props.pointerEvents,
    position: props.position,
    right: props.right,
    top: props.top,
    width: props.fit ? '100%' : props.width,
    WebkitOverflowScrolling: props.overflowScrolling,
    zIndex: props.zIndex,
  }
}

function hasHitSlopProp(props: IProps) {
  return props.hitSlop
    || props.hitSlopVertical
    || props.hitSlopHorizontal
    || props.hitSlopTop
    || props.hitSlopRight
    || props.hitSlopBottom
    || props.hitSlopLeft
}

const Slop = (props: IProps) => (
  <span
    style={{
      position: 'absolute',
      top: -(props.hitSlopTop || props.hitSlopVertical || props.hitSlop || 0),
      right: -(props.hitSlopRight || props.hitSlopHorizontal || props.hitSlop || 0),
      bottom: -(props.hitSlopBottom || props.hitSlopVertical || props.hitSlop || 0),
      left: -(props.hitSlopLeft || props.hitSlopHorizontal || props.hitSlop || 0),
    }}
  />
)

export default class View extends React.PureComponent<IProps, void> {
  static defaultProps = {
    tag: 'div',
    shrink: 0,
    position: 'relative',
    alignContent: 'flex-start',
  }

  render() {
    const styleFromProps = getStyleFromProps(this.props)
    const propsToPass = _omit(this.props, propsToOmit)

    const css = { ...styleFromProps, ...this.props.style }
    propsToPass.css = css

    // inlineStyle should pass down as style to apply inline
    propsToPass.style = this.props.inlineStyle

    // Use refNode pattern to pass back the DOM's node
    if (this.props.refNode) {
      propsToPass.ref = this.props.refNode
    }

    if (hasHitSlopProp(this.props)) {
      const HitSlop = (
        <Slop
          hitSlop={this.props.hitSlop}
          hitSlopTop={this.props.hitSlopTop}
          hitSlopRight={this.props.hitSlopRight}
          hitSlopBottom={this.props.hitSlopBottom}
          hitSlopLeft={this.props.hitSlopLeft}
          hitSlopVertical={this.props.hitSlopVertical}
          hitSlopHorizontal={this.props.hitSlopHorizontal}
        />
      )

      if (propsToPass.children == null) {
        propsToPass.children = HitSlop
      }
      else {
        // convert children to array, then prepend Slop
        propsToPass.children = React.Children.toArray(propsToPass.children)
        propsToPass.children.unshift(HitSlop)
      }
    }

    return glamorReact.createElement(this.props.tag, propsToPass)
  }
}