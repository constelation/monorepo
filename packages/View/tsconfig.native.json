{
    "compilerOptions": {
        "module": "es2015",
        "target": "es2015",
        "jsx": "react-native",
        "outDir": ".",
        "moduleResolution": "node",
        "removeComments": true,
        "allowSyntheticDefaultImports": true,
        "noImplicitAny": true,
        "experimentalDecorators": true,
        "preserveConstEnums": true,
        "allowJs": false,
        "declaration": true
    },
    "files": [
        "src/index.native.tsx"
    ]
}