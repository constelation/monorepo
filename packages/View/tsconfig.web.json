{
    "compilerOptions": {
        // "module": "es6",
        "module": "commonjs",
        "target": "es5",
        "lib": [
            "dom",
            "es5",
            "es2015.promise"
        ],
        "jsx": "react",
        "outDir": ".",
        "removeComments": true,
        "allowSyntheticDefaultImports": true,
        "noImplicitAny": true,
        "experimentalDecorators": true,
        "preserveConstEnums": true,
        "allowJs": false,
        "declaration": true
    },
    "files": [
        "src/index.tsx"
    ]
}