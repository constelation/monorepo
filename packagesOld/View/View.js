'use strict';

var flexComponentFactory = require('constelation-flex-component-factory')

var requiredStyle = {
  flexDirection: 'column',
}

var styleAliases = {
  alignHorizontal: {
    property: 'alignItems',
    map: {
      left: 'flex-start',
      center: 'center',
      right: 'flex-end',
    },
  },
  alignVertical: {
    property: 'justifyContent',
    map: {
      top: 'flex-start',
      center: 'center',
      bottom: 'flex-end',
    },
  },
}

// export default displayComponentFactory('View', style, defaultStyle, styleAliases)
module.exports = flexComponentFactory('View', requiredStyle, undefined, styleAliases)
