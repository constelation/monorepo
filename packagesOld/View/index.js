// export { default } from './View'
var View = require('./View')

module.exports = View
