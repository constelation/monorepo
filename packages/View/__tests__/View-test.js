// import 'react-native'
import React from 'react'
import View from '../index.js'
import {shallow} from 'enzyme';
import { render, unmountComponentAtNode } from 'react-dom'
import { cssLabels, rehydrate } from 'glamor'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <View />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
// test('alignsVertical', () => {
//   // render(<View alignVertical='top' />, node, () => {
//   //   expect(childStyle(node).justifyContent).toEqual('flex-start')
//   // })
//
//   cssLabels(true)
//   const Comp = shallow(<View alignVertical='top'/>)
//
//   const className = Object.keys(Comp.prop('className'))[0]
//   console.log(Comp.props());
//     // console.log(rehydrate([className]));
//   // console.log(Comp.prop('className'));
//   // expect(Comp.prop('style').justifyContent).toBe('flex-start')
//   expect(Comp.instance().props.style.justifyContent).toBe('flex-start')
//   cssLabels(false)
//   // expect(tree).toMatchSnapshot();
// });
//
//
// // it('applies the style to a given node', () => {
// //     render(<div {...style({ backgroundColor: 'red' })}/>, node, () => {
// //       expect(childStyle(node).backgroundColor).toEqual('rgb(255, 0, 0)')
// //     })
// //   })
// //
// function childStyle(node, p = null) {
//   return window.getComputedStyle(node.childNodes[0], p)
// }
