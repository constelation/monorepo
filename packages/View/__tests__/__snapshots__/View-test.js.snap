exports[`test renders correctly 1`] = `
<div
  className={
    Object {
      "data-css-1dhhihx": "",
    }
  }
  srink={0}
  style={undefined} />
`;
